src/udiskslinuxpartition.c

```diff
@@ -812,13 +847,29 @@ handle_resize (UDisksPartition       *partition,
       goto out;
     }
 
+  /* Wait for partition to reappear before returning, so that clients
+   * cat at least issue udisks_client_settle() afterwards to wait for
+   * the other interfaces to be present. If the resizing/fitting results
+   * in the same partition size this waiter will hit its timeout
+   * because there is no way to distinguish the reappeared partition
+   * from the original one.
+   */
```

Comment:
We know both the original size and the new requested size. Why don't we just skip the waiting below if those are the same? Or just wait for the partition object instead (if it disappears for a while)?